lxd/device/pci/pci.go

```diff
@@ -158,3 +158,32 @@ func deviceProbeWait(pciDev Device) error {
 
 	return fmt.Errorf("Device took too long to activate at %q", driverPath)
 }
+
+// NormaliseAddress converts common PCI address notation to the kernel's notation.
+func NormaliseAddress(addr string) string {
+	// Ignore empty values.
+	if addr == "" {
```

Comment:
@stgraber I would prefer this function to not have device specific logic in it (as the new pci device wouldn't accept an empty device string).